server/pulp/server/managers/status.py

```diff
@@ -51,9 +54,14 @@ def get_broker_conn_status():
     # explicit connection and then release it.
     # See https://github.com/celery/kombu/issues/432 for more detail.
     try:
-        conn = celery.connection()
         conn.connect()
+        transport = conn._transport
         conn.release()
+
+        # An explicit call to __del__ to release file descriptors
+        # https://pulp.plan.io/issues/2124#note-9
+        transport.__del__()
```

Comment:
I was talking in `#python` and they explained that I should never expect the code in `__del__()` to be run. I think that is where we're going wrong with these file descriptor leaks. There is an improvement to be made here, but I think it's in kombu by adjusting the way the kombu.Connection.Transport closes itself [here](https://github.com/celery/kombu/blob/4401770d8b007ae164a833450d4b27252b58c3d6/kombu/connection.py#L309). This should have some sort of callback to release resources. For instance there is a `collect()` method [here](https://github.com/celery/kombu/blob/4401770d8b007ae164a833450d4b27252b58c3d6/kombu/connection.py#L313-L334), but I'm not sure if its appropriate for us to call that here instead or what that does for other cases of connection error.

Aside from this, observing the file descriptor count when there is a connection error would be illustrative also. Maybe I will try that too.